google/cloud/spanner/row_parser.h

```diff
@@ -0,0 +1,165 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_SPANNER_ROW_PARSER_H_
+#define GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_SPANNER_ROW_PARSER_H_
+
+#include "google/cloud/spanner/row.h"
+#include "google/cloud/spanner/value.h"
+#include "google/cloud/spanner/version.h"
+#include "google/cloud/status.h"
+#include "google/cloud/status_or.h"
+#include <array>
+#include <tuple>
+#include <utility>
+
+namespace google {
+namespace cloud {
+namespace spanner {
+inline namespace SPANNER_CLIENT_NS {
+
+/**
+ * A RowParser takes a range of `Value` objects and a list of types, and it
```

Comment:
I would expect you need to receive a range of `StatusOr<Value>` objects? The "unchunker" is likely to produce ranges of `StatusOr<Value>` because the iteration might fail?

Reply:
That, uhh, might be true? I'm not sure yet exactly what the connection will be between the RowParser and the stream of [PartialResultSet](https://github.com/googleapis/googleapis/blob/master/google/spanner/v1/result_set.proto#L61) will be. Yes, something will need to unchunk it, and I'm not sure if there's another way to expose Status for a failed unchunking.

But it should be easy to adapt this code to accept an input range of `StatusOr<Value>` instead (I think).